bin/semantic-release.js

```diff
@@ -1,165 +1,31 @@
 #!/usr/bin/env node
 
-var fs = require('fs')
-var path = require('path')
-var url = require('url')
+// Bad news: We have to write plain ES5 in this file
+// Good news: It's the only file of the entire project
 
-var _ = require('lodash')
-var log = require('npmlog')
-var nopt = require('nopt')
-var npmconf = require('npmconf')
-var normalizeData = require('normalize-package-data')
+var semver = require('semver')
 
-log.heading = 'semantic-release'
-var env = process.env
-var pkg = JSON.parse(fs.readFileSync('./package.json'))
-var originalPkg = _.cloneDeep(pkg)
-normalizeData(pkg)
-var knownOptions = {
-  branch: String,
-  debug: Boolean,
-  'github-token': String,
-  'github-url': String,
-  'analyze-commits': [path, String],
-  'generate-notes': [path, String],
-  'verify-conditions': [path, String],
-  'verify-release': [path, String]
-}
-var options = _.defaults(
-  _.mapKeys(nopt(knownOptions), function (value, key) {
-    return _.camelCase(key)
-  }),
-  pkg.release,
-  {
-    branch: 'master',
-    fallbackTags: {
-      next: 'latest'
-    },
-    debug: !env.CI,
-    githubToken: env.GH_TOKEN || env.GITHUB_TOKEN,
-    githubUrl: env.GH_URL
-  }
-)
-var plugins = require('../src/lib/plugins')(options)
-
-npmconf.load({}, function (err, conf) {
-  if (err) {
-    log.error('init', 'Failed to load npm config.', err)
-    process.exit(1)
-  }
-
-  var npm = {
-    auth: {
-      token: env.NPM_TOKEN
-    },
-    cafile: conf.get('cafile'),
-    loglevel: conf.get('loglevel'),
-    registry: require('../src/lib/get-registry')(pkg, conf),
-    tag: (pkg.publishConfig || {}).tag || conf.get('tag') || 'latest'
-  }
-
-  // normalize trailing slash
-  npm.registry = url.format(url.parse(npm.registry))
-
-  log.level = npm.loglevel
-
-  var config = {
-    env: env,
-    pkg: pkg,
-    options: options,
-    plugins: plugins,
-    npm: npm
-  }
-
-  var hide = {}
-  if (options.githubToken) hide.githubToken = '***'
-
-  log.verbose('init', 'options:', _.assign({}, options, hide))
-  log.verbose('init', 'Verifying config.')
-
-  var errors = require('../src/lib/verify')(config)
-  errors.forEach(function (err) {
-    log.error('init', err.message + ' ' + err.code)
-  })
-  if (errors.length) process.exit(1)
-
-  if (options.argv.remain[0] === 'pre') {
-    log.verbose('pre', 'Running pre-script.')
-    log.verbose('pre', 'Veriying conditions.')
+if (semver.lt(process.version, '8.0.0')) {
+  console.error(
+`semantic-release: node version >= 8 is required. Found ${process.version}.
 
-    plugins.verifyConditions(config, function (err) {
-      if (err) {
-        log[options.debug ? 'warn' : 'error']('pre', err.message)
-        if (!options.debug) process.exit(1)
-      }
+Don't fret. It's not that big of a deal.
```

Comment:
I know people will have strong opinions about. I know this is meant to sound fun & human, but it just provoke heated emotions. I would suggest the information super dry here, the argument is solid, no need to tell anyone not to fret, I’d just go straight into a simple, clear explanation.